Jenkinsfile

```diff
@@ -1,7 +1,7 @@
 #!/usr/bin/env groovy
 
 
-def tested_files = "build/.*|deploy/.*|roles/.*|tests/smoketest/.*|Makefile|watches.yaml|Jenkinsfile"
+def tested_files = "\"build/.\\*|deploy/.\\*|roles/.\\*|tests/smoketest/.\\*|Makefile|watches.yaml|Jenkinsfile\""
```

Comment:
Did this actually fix it, or does it have the failure to detect changes on new branches as you described?

Reply:
That issue is fixed, but this isn't the line that fixed it. The condition before was just all wrong, it compared changes only with the previous build. That's why it worked for me when testing (I had some builds before I started working on it) and that's why it didn't work when creating a PR (there weren't any previous builds).

Now I made it actually look at the target branch of the PR and compare the changes with that, so it doesn't care about previous builds.